modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py

```diff
@@ -128,22 +129,49 @@ def read_csv(cls, **kwargs):  # noqa: PR01
             if not use_modin_impl:
                 raise ArrowEngineException(error_message)
 
-            dtype = kwargs["dtype"]
-            if isinstance(dtype, dict):
-                column_types = {c: cls._dtype_to_arrow(t) for c, t in dtype.items()}
-            else:
-                column_types = cls._dtype_to_arrow(dtype)
-
-            parse_dates = kwargs["parse_dates"]
-            if (type(parse_dates) is list) and type(column_types) is dict:
-                for c in parse_dates:
-                    column_types[c] = pa.timestamp("s")
-
-            names = kwargs["names"]
+            if (names := kwargs["names"]) is lib.no_default:
+                names = None
             skiprows = kwargs["skiprows"]
-            if names not in (lib.no_default, None) and kwargs["header"] == 0:
+            if names and kwargs["header"] == 0:
                 skiprows = skiprows + 1 if skiprows is not None else 1
 
+            column_names = None
+
+            def get_col_names():
+                nonlocal column_names
+                if column_names is None:
+                    # Using pandas to read the column names
+                    column_names = pandas.read_csv(
+                        kwargs["filepath_or_buffer"], nrows=0, engine="c"
+                    ).columns.tolist()
+                return column_names
+
+            if dtype := kwargs["dtype"]:
```

Comment:
Can we have maybe brief comment explaining the logic behind dtype/parse_dates cases we're handling here

Reply:
Done.